### PR TITLE
[INTEL MKL] Fix control_flow_ops_test unit test failure

### PR DESCRIPTION
**Pull request by gzmkl**
This PR fixes UT failure control_flow_ops_test, which is related to MaxPool3DGrad.

MaxPool3DGrad should only be rewritten when workspace is available.

### CODE REVIEW

File: tensorflow/core/common_runtime/mkl_layout_pass.cc [hunk 1]
```diff
@@ -1694,6 +1694,24 @@ class MklLayoutRewritePass : public GraphOptimizationPass {
     return do_rewrite;
   }
 
+  static bool Maxpool3DGradRewrite(const Node* n) {
+    CHECK_NOTNULL(n);
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass.cc:1698]: Nit: We don't use `CHECK*` in new code anymore.
```suggestion
    DCHECK(n);
```

File: tensorflow/core/common_runtime/mkl_layout_pass.cc [hunk 2]
```diff
@@ -1694,6 +1694,24 @@ class MklLayoutRewritePass : public GraphOptimizationPass {
...
+    bool do_rewrite = false;
+    for (const Edge* e : n->in_edges()) {
+      // Rewrite only if there is corresponding Maxpool3D, i.e., workspace is
+      // available
+      if (e->dst()->type_string() == csinfo_.max_pool3d_grad &&
+          e->dst_input() == 1 &&
+          e->src()->type_string() ==
+              mkl_op_registry::GetMklOpName(csinfo_.max_pool3d) &&
+          e->src_output() == 0) {
+        do_rewrite = true;
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass.cc:1708]: Nit: We can just `return true;` here and `return false;` outside the `for` loop. (Don't need `do_rewrite`.)

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! I have two minor comments.

**Comment by ShengYang1** [comment]
@googlebot I consent.

**Comment by ShengYang1** [comment]
@penpornk Thanks for reviewing. I have made some changes.

File: tensorflow/core/common_runtime/mkl_layout_pass.cc (hunk 1 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/mkl_layout_pass.cc]: I have replaced all "CHECK_NOTNULL" usage with "DCHECK" in the same source file - there are ~20. 

I will push the changes once local test is done.

**Review by penpornk — APPROVED** [review]
Thank you again for the PR!

File: tensorflow/core/common_runtime/mkl_layout_pass.cc (hunk 2 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/mkl_layout_pass.cc]: The suggestion is much better. I will do the change.

File: tensorflow/core/common_runtime/mkl_layout_pass.cc (hunk 1 above)
Comment (penpornk) [reply tensorflow/core/common_runtime/mkl_layout_pass.cc]: Thank you! Just for future reference, it's not required change `CHECK*` in old code. We just try not to have `CHECK*` in any part that is changed/added by the PR. :)

**Review by penpornk — APPROVED** [review]
Thank you again for the changes!

**Review by rthadur — APPROVED** [review]

**Review by penpornk — APPROVED** [review]
Thank you for merging conflicts!

**Pull request merged**